EngineLayer/CrosslinkSearch/CrosslinkSpectralMatch.cs

```diff
@@ -109,6 +112,43 @@ public static List<int> GetPossibleCrosslinkerModSites(char[] crosslinkerModSite
             return possibleXlPositions;
         }
 
+        /// <summary>
+        /// The beta peptide is not added to the peptide dictionary of PSM. So if we want the longest ion series, we have to get it elsewhere.
+        /// </summary>
+        /// <param name="peptide"></param>
+        /// <returns></returns>
+        public int GetLongestIonSeriesBidirectionalBetaPeptide(List<MatchedFragmentIon> matchedFragmentIons, PeptideWithSetModifications peptide)
```

Comment:
This is similar to PSM.GetLongestIonSeriesBidirectional. combine it.

Reply:
done